src/modules/user/user.controller.ts

```diff
@@ -87,4 +99,28 @@ export class UserController {
   ) {
     return this.userService.getUsersByAdmin(page, limit, req.user);
   }
+
+  @Patch(':userId/status')
+  @ApiOperation({ summary: 'Update a user status (Super Admin only)' })
+  @ApiOkResponse({ description: 'Status updated successfully', type: UpdateUserStatusResponseDto })
+  @ApiUnauthorizedResponse({
+    description: 'User is not authorized',
+    type: 'object',
+    example: {
+      message: 'User is currently unauthorized, kindly authenticate to continue',
+      status: 401,
+    },
+  })
+  @ApiForbiddenResponse({
+    description: 'User is forbidden',
+    example: {
+      message: 'You dont have the permission to perform this action',
+      status: 403,
+    },
+  })
+  @ApiInternalServerErrorResponse({ description: 'Internal Server Error' })
```

Comment:
Internal server error is not something you want your client to see. Please remove